libbeat/kibana/index_pattern_generator.go

```diff
@@ -11,43 +10,56 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
-type Index struct {
-	Version   string
-	IndexName string
-	BeatDir   string
-	BeatName  string
-
+type IndexPatternGenerator struct {
+	indexName        string
+	version          string
 	fieldsYaml       string
 	targetDirDefault string
 	targetDir5x      string
 	targetFilename   string
 }
 
-// Create the Index-Pattern for Kibana for 5.x and default.
-func (i *Index) Create() ([]string, error) {
-	indices := []string{}
+// Create an instance of the Kibana Index Pattern Generator
+func NewGenerator(indexName, beatName, beatDir, version string) (IndexPatternGenerator, error) {
```

Comment:
We normally return a reference here, this also make it possible to return `nil, err` on line 28.